static/src/javascripts/lib/time-utils.ts

```diff
@@ -0,0 +1,16 @@
+// from and to should be Epoch time in milliseconds
+const dateDiffDays = (from: number, to: number): number => {
+	const oneDayMs = 1000 * 60 * 60 * 24;
+	const diffMs = to - from;
+	return Math.floor(diffMs / oneDayMs);
+};
+
+const isExpired = (testExpiry: string): boolean => {
+	// new Date(test.expiry) sets the expiry time to 00:00:00
+	// Using SetHours allows a test to run until the END of the expiry day
+	const startOfToday = new Date().setHours(0, 0, 0, 0);
+	const expiryDate = new Date(testExpiry).getTime();
```

Comment:
Calling `getTime` here is the only difference in the logic between the typescript and javascript versions of this file.
In the js version, we were comparing a number (`startOfToday`) to a date object (`expiryDate`) and after the change we're comparing two numbers.